File: 50 Coding Challenge Part I/TypeScript/Challenge22.ts
// Challenge 22: Reverse an array

function reverseArray (arr: number[]): number[] {
    let newArr: number[] = [];

    for (let i = arr.length - 1; i >= 0; i--){
        newArr.push(arr[i]);
    }

    return newArr;
}

// Driver Method
let arr = [1, 2, 3, 4, 5, 6];
console.log(reverseArray(arr));

File: 50 Coding Challenge Part I/dart/lib/challenge22.dart
/*  
  Coding Challenge 22: Reverse an array

  @author: mofirojean
*/

List<int> reverseArray(List<int> nums) {
  List<int> reverse = [];
  for (var i = nums.length - 1; i >= 0; i--) {
    reverse.add(nums[i]);
  }
  return reverse;
}
